src/dhcpmon/src/dhcp_device.c

```diff
@@ -152,6 +196,45 @@ static void handle_dhcp_option_53(dhcp_device_context_t *context,
     }
 }
 
+/**
+ * @code handle_dhcpv6_option(context, dhcp_option, dir);
+ *
+ * @brief handle the logic related to DHCPv6 option
+ *
+ * @param context       Device (interface) context
+ * @param dhcp_option   pointer to DHCP option buffer space
+ * @param dir           packet direction
+ *
+ * @return none
+ */
+static void handle_dhcpv6_option(dhcp_device_context_t *context,
+                                  const u_char dhcp_option,
+                                  dhcp_packet_direction_t dir)
+{
+    switch (dhcp_option)
+    {
+    case DHCPv6_MESSAGE_TYPE_SOLICIT:
+    case DHCPv6_MESSAGE_TYPE_REQUEST:
+    case DHCPv6_MESSAGE_TYPE_CONFIRM:
+    case DHCPv6_MESSAGE_TYPE_RENEW:
+    case DHCPv6_MESSAGE_TYPE_REBIND:
+    case DHCPv6_MESSAGE_TYPE_RELEASE:
+    case DHCPv6_MESSAGE_TYPE_DECLINE:
+    case DHCPv6_MESSAGE_TYPE_ADVERTISE:
+    case DHCPv6_MESSAGE_TYPE_REPLY:
+    case DHCPv6_MESSAGE_TYPE_RECONFIGURE:
+    case DHCPv6_MESSAGE_TYPE_INFORMATION_REQUEST:
+    case DHCPv6_MESSAGE_TYPE_RELAY_FORWARD:
```

Comment:
can we instead of counting relay forward and relay reply to count those packet against the underlying type with dir set as :
relay-forward: TX dir of client packets
relay-reply: RX dir of server packets.

Reply:
The current implementation is counting with the direction.
On the condition to check if health status is ok, we check if solicit/request packets received and then check if relay-forward packet transmitted, can you elaborate how is it possible to simplify this check?

Reply:
Addressed on a different PR.